h/services/nipsa.py

```diff
@@ -70,13 +70,18 @@ def unflag(self, user):
         user.nipsa = False
         if self._flagged_userids is not None:
             self._flagged_userids.remove(user.userid)
-        reindex_user_annotations.delay(user.userid)
+        self._reindex_users_annotations(user, tag="NipsaService.unflag()")
 
     def clear(self):
         """Unload the cache of flagged userids, if populated."""
         self._flagged_userids = None
 
+    def _reindex_users_annotations(self, user, tag):
+        self._search_index.add_users_annotations(
+            user.userid, tag=tag, force=True, schedule_in=30
+        )
```

Comment:
We've changed `user.nipsa` in the DB but that won't be committed until the request finishes and commits its transaction. So we schedule the reindex to start no sooner than 30 seconds from now, to avoid a race condition with the DB commit

Reply:
Well, we'll see I guess. But the rename user admin page can also use `_search_index.add_users_annotations()` and won't necessarily need `force=True`. I can see a need for a future "reindex user's annotations" admin page as well